src/validations/html/rules/MismatchedTag.js

```diff
@@ -0,0 +1,139 @@
+import {List, Map, Record, Set} from 'immutable';
```

Comment:
(Also I’m pretty skeptical that `Immutable.List` would provide any tangible performance benefits over `Array` for the queue use case here, although I’m certainly down to be proven wrong with a jsperf etc.)